runtime/src/main/java/org/corfudb/runtime/view/ISealReplicationView.java

```diff
@@ -0,0 +1,35 @@
+package org.corfudb.runtime.view;
+
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.QuorumUnreachableException;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Seals all the log unit servers in every stripe in the specified segment following a certain replication method.
+ *
+ * Created by zlokhandwala on 3/7/17.
+ */
+public interface ISealReplicationView {
```

Comment:
This interface seems to be the cause of a lot of redundant code in this pull request.
One alternative to simplify(i.e. remove a lot of this code) would be to use helper methods instead of this interface and its implementations.

For example, you can do something like this:

Set<CompletableFuture> serversToSeal(Set<Addresses> servers) 
Set<CompletableFuture> waitForSeal(Set<CompletableFuture> servers, Timeout miliseconds) (this method will return futures for all servers that havent completed by the end of the timeout)

So then for chain replication you can have:

Set<CompletableFuture> servers = (set of servers to seal);
Set<CompletableFuture> allServers  = serversToSeal(servers);
Set<CompletableFuture> unsealedServers =  waitForSeal(Set<CompletableFuture> servers, infinite timeout)

And for chain replication, you would have a reasonable timeout set instead of infinity.